tests/iostability/test_iostability_degraded_path.py

```diff
@@ -84,41 +85,50 @@ def setup_class(cls):
         cls.s3t_obj = S3TestLib()
         cls.ha_obj = HAK8s()
         cls.s3userinfo = {'accesskey': ACCESS_KEY, 'secretkey': SECRET_KEY}
+        cls.remote_dir_path = cls.test_cfg['remote_path']
 
     def setup_method(self):
         """Setup Method"""
         self.log.info("Setup Method Started")
         self.log.info("Start Procpath collection")
         self.proc_path = EnableProcPathStatsCollection(CMN_CFG)
         self.log_collect = ServerOSLogsCollectLib(CMN_CFG)
+        self.top_stats = TopStatsCollection(CMN_CFG)
         resp = self.proc_path.setup_requirement()
         assert_utils.assert_true(resp[0], resp[1])
         self.proc_path.start_collection()
         time.sleep(30)
         resp = self.proc_path.validate_collection()
         assert_utils.assert_true(resp[0], resp[1])
+        resp = self.top_stats.collect_stats(dir_path=self.remote_dir_path)
+        assert_utils.assert_true(resp)
         self.test_completed = False
         self.log.info("Setup Method Ended")
 
     def teardown_method(self):
         """Teardown method."""
         self.log.info("Teardown method")
+        path = os.path.join(LOG_DIR, LATEST_LOG_FOLDER)
         if not self.test_completed:
             self.mail_notify.event_fail.set()
             self.log.info("Test Failure observed, collecting support bundle")
-            path = os.path.join(LOG_DIR, LATEST_LOG_FOLDER)
             resp = support_bundle_utils.collect_support_bundle_k8s(local_dir_path=path,
                                                                    scripts_path=K8S_SCRIPTS_PATH)
             assert_utils.assert_true(resp)
             resp = self.log_collect.collect_logs(path=path)
             assert_utils.assert_true(resp)
         else:
             self.mail_notify.event_pass.set()
+            self.top_stats.stop_collection()
         self.log.info("Stop Procpath collection")
         self.proc_path.stop_collection()
         self.log.info("Copy files to client")
         resp = self.proc_path.get_stat_files_to_local()
         self.log.debug("Resp : %s", resp)
+        resp = self.top_stats.stop_collection(dir_path=self.remote_dir_path)
```

Comment:
not able to understand why do we need stop_collection at 2 places? line 122 and 128? We need to stop in both cases so do we really need at line 122?

Reply:
will remove 122 line.